_posts/2024-01-03-#28956.md

```diff
@@ -7,20 +7,58 @@ authors: [dergoegge]
 components: ["p2p", "consensus"]
 host: stickies-v
 status: upcoming
-commit:
+commit: 87c577e9742d7154826c755a7fe320f34fd54c81
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- Bitcoin relies on the concept of time for operations such as the difficulty adjustment and transaction- or script-level timelocks. Since time is relative and clocks are notoriously difficult to keep synchronized, it is impossible to define a source of truth that does not depend on synchronization and authorities such as NTP servers. In a decentralized system, we must accept that nodes can have a different but equally valid view of the current time.
+
+- The Bitcoin network comes to consensus on time by requiring miners to commit to a timestamp in the [block header](https://github.com/bitcoinbook/bitcoinbook/blob/develop/ch09.asciidoc#block-header). To prevent miners from being able to use excessively deviating timestamps in their blocks, nodes [verify](https://github.com/bitcoin/bitcoin/blob/3a0f54dd2402d74a5ac4304b3ad09014cfb25edf/src/validation.cpp#L3803-L3810) that the timestamp is later than the `Median Time Past` (calculated on the previous 11 blocks), and earlier than 2 hours from the current time.
+
+- Prior to this PR, the current time was calculated based on the network-adjusted time. During the handshake, the peer node's current time is compared to the system's current time, and the time difference is [stored](https://github.com/bitcoin/bitcoin/blob/3a0f54dd2402d74a5ac4304b3ad09014cfb25edf/src/net_processing.cpp#L3556). The network-adjusted time is then calculated by adding the median of these offsets to the system's current time, unless it exceeds the `-maxtimeadjustment` value [in which case a warning is emitted](https://github.com/bitcoin/bitcoin/blob/3a0f54dd2402d74a5ac4304b3ad09014cfb25edf/src/timedata.cpp#L84-L100).
```

Comment:
This could clarify what is peer-specific and how that becomes our view of the network.

```suggestion
- Prior to this PR, the current time was calculated based on the network-adjusted time. When connecting to a peer, the peer's current time is compared to the system's current time, and the time difference is [stored](https://github.com/bitcoin/bitcoin/blob/3a0f54dd2402d74a5ac4304b3ad09014cfb25edf/src/net_processing.cpp#L3556). The network-adjusted time is then calculated by adding the median of all peers' offsets to the system's current time, unless it exceeds the `-maxtimeadjustment` value [in which case a warning is emitted](https://github.com/bitcoin/bitcoin/blob/3a0f54dd2402d74a5ac4304b3ad09014cfb25edf/src/timedata.cpp#L84-L100).
```

Reply:
Thanks, I'll include this when I update the notes after the PR is rebased!

Reply:
Included in https://github.com/bitcoin-core-review-club/website/pull/738